files/Aboot/boot0.j2

```diff
@@ -657,22 +658,53 @@ write_default_cmdline() {
     cmdline_clear
 
     if $in_aboot; then
-        # generate the default kernel parameters for the platform
+        # Generate the default kernel parameters for the platform
         cat /etc/cmdline | sed "/^\(${bootconfigvars// /\|}\|crashkernel\|loglevel\|ignore_loglevel\)\(\$\|=\)/d;/^\$/d" | cmdline_append
-    elif grep -q "$delimiter" /proc/cmdline; then
-        # we are on a recent sonic image using delimiter. extracting the part of the
+    elif grep -q "$delimiter" /proc/cmdline && ! grep -Eq "log_buf_len=.* $delimiter" /proc/cmdline; then
```

Comment:
Is it always going to be safe to depend on `log_buf_len=.* $delimiter` to detect append bug? I mean hardcoding this check to look for `log_buf_len` can possibly lead to false result?

Reply:
You are making a good point since I just recalled that log_buf_len was added recently, I should use `varlog_size` instead that has been here since the beginning.
It shouldn't lead to false result since these parameters are never set by Aboot, and should therefore never be before `cmdline-aboot-end`